src/serde/read_cache_lookup.rs

```diff
@@ -147,11 +187,11 @@ impl ReadCacheLookup {
                     for (parent, direction) in items.iter() {
                         if *(self.count.get(parent).unwrap_or(&0)) > 0 && !seen_ids.contains(parent)
                         {
-                            let mut new_path = path.clone();
-                            new_path.push(*direction);
-                            if new_path.len() > max_path_length {
+                            if path.len() + 1 > max_path_length {
```

Comment:
how about using `>=` rather then `+1`?

Reply:
And we're sure this doesn't change anything? Oh I see, the `new_path` stuff is deferred until after this line. Okay, seems good.

Reply:
yeah, I don't think this makes a huge difference. I don't think it happens very often, I was just on the look-out for avoiding heap allocations.

Reply:
I think I like `+ 1` a bit better, since we're checking the state *as-if* we had added the new path